### PR TITLE
Skip building a source map when parsing NaCl unless it is really needed

### PR DESCRIPTION
**Pull request by ori-moisis**
Building a source map during NaCl parse makes parsing significantly slower.
It also greatly increases memory consumption.
It is also not actually needed in most flows and it is actually usually discarded.

Exposing the internal option to skip generating a source map and using it to improve overall perforamnce
when loading a workspace without a cache

---

Almost all flows that use parse actually don't need the source map.
However, since generating it was the default and this is an exported function that can be used in other projects, I kept the default behavior as-is for now and only skipped the source map where it was especially costly.
In a future PR we will change this default (just need to make sure all external users of this api that need a source map pass that option explicitly)

---
_Release Notes_: 
Core:
- Improve workspace loading times and memory consumption when working with out-dated cache

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/workspace/src/parser/parse.ts [hunk 1]
```diff
@@ -47,9 +51,10 @@ export const parse = async (
   naclFile: Buffer,
   filename: string,
   functions: Functions = {},
+  options?: ParseOptions,
 ): Promise<Required<ParseResult>> => {
   const srcString = naclFile.toString()
-  return parseBuffer(srcString, filename, functions)
```
Comment (adamro) [inline packages/workspace/src/parser/parse.ts:52]: Do we plan to install a beautiful AC here? :D [why is this not a single named boolean / 2 fn's if there's only one option]

Comment (ori-moisis) [reply packages/workspace/src/parser/parse.ts:52]: how do you create a single **named** boolean?
I don't want to have a call to `parse(..., true)` because that becomes very hard to follow very quickly (what is "true"?)
and I didn't want to completely break the interface and make everything a named parameter since this is an exported function.
so this was the compromise

**Comment by ori-moisis** [comment]
will reopen when I get to fixing this
